src/common/meta/src/cache/flow/table_flownode.rs

```diff
@@ -0,0 +1,202 @@
+
+use std::collections::HashSet;
+use std::sync::Arc;
+
+use futures::future::BoxFuture;
+use futures::TryStreamExt;
+use moka::future::Cache;
+use moka::ops::compute::Op;
+use table::metadata::TableId;
+
+use crate::cache::{CacheContainer, Initializer};
+use crate::error::Result;
+use crate::instruction::{CacheIdent, CreateFlow, DropFlow};
+use crate::key::flow::{TableFlowManager, TableFlowManagerRef};
+use crate::kv_backend::KvBackendRef;
+use crate::FlownodeId;
+
+type FlownodeSet = HashSet<FlownodeId>;
+
+/// [TableFlownodeSetCache] caches the [TableId] to [FlownodeSet] mapping.
+pub type TableFlownodeSetCache = CacheContainer<TableId, FlownodeSet, CacheIdent>;
+
+/// Constructs a [TableFlownodeSetCache].
+pub fn new_table_flownode_set_cache(
+    name: String,
+    cache: Cache<TableId, FlownodeSet>,
+    kv_backend: KvBackendRef,
+) -> TableFlownodeSetCache {
+    let table_flow_manager = Arc::new(TableFlowManager::new(kv_backend));
+    let init = init_factory(table_flow_manager);
+
+    CacheContainer::new(name, cache, Box::new(invalidator), init, Box::new(filter))
+}
+
+fn init_factory(table_flow_manager: TableFlowManagerRef) -> Initializer<TableId, FlownodeSet> {
+    Arc::new(move |&table_id| {
+        let table_flow_manager = table_flow_manager.clone();
+        Box::pin(async move {
+            table_flow_manager
+                .flows(table_id)
+                .map_ok(|key| key.flownode_id())
+                .try_collect::<HashSet<_>>()
+                .await
+                .map(Some)
+        })
+    })
+}
+
+async fn invalidate_create_flow(
+    cache: &Cache<TableId, FlownodeSet>,
+    CreateFlow {
+        source_table_ids,
+        flownode_ids,
+    }: &CreateFlow,
+) {
+    for table_id in source_table_ids {
+        let entry = cache.entry(*table_id);
+        entry
+            .and_compute_with(
+                async |entry: Option<moka::Entry<u32, HashSet<u64>>>| match entry {
+                    Some(entry) => {
+                        let mut set = entry.into_value();
+                        set.extend(flownode_ids.clone());
+
+                        Op::Put(set)
+                    }
+                    None => Op::Put(HashSet::from_iter(flownode_ids.clone())),
+                },
+            )
+            .await;
+    }
+}
+
+async fn invalidate_drop_flow(
+    cache: &Cache<TableId, FlownodeSet>,
+    DropFlow {
+        source_table_ids,
+        flownode_ids,
+    }: &DropFlow,
+) {
+    for table_id in source_table_ids {
+        let entry = cache.entry(*table_id);
+        entry
+            .and_compute_with(
+                async |entry: Option<moka::Entry<u32, HashSet<u64>>>| match entry {
+                    Some(entry) => {
+                        let mut set = entry.into_value();
+                        for flownode_id in flownode_ids {
+                            set.remove(flownode_id);
+                        }
+
+                        Op::Put(set)
```

Comment:
Does it need to remove the entry if the set is empty?

Reply:
No, if remove the entry, it will fetch from remote while next invoking.